src/components/App.jsx

```diff
@@ -112,23 +114,16 @@ export default class App extends React.Component {
 
   onStyleChanged(newStyle, save=true) {
 
-    if(newStyle.glyphs === undefined){
-      let error = "Failed because no glyphs property found in style";
-      this.setState({
-        errors:[error]
-      })
-      return
-    }
-
-    if(newStyle.glyphs !== this.state.mapStyle.glyphs) {
-      this.updateFonts(newStyle.glyphs)
-    }
-    if(newStyle.sprite !== this.state.mapStyle.sprite) {
-      this.updateIcons(newStyle.sprite)
-    }
-
     const errors = styleSpec.validate(newStyle, styleSpec.latest)
     if(errors.length === 0) {
+
+      if(newStyle.glyphs !== this.state.mapStyle.glyphs) {
```

Comment:
Hmmm, I wonder if it's maybe it's safer to just do

```
if(newStyle.hasOwnProperty("glyphs") && newStyle.glyphs !== this.state.mapStyle.glyphs) { ... }
```

...and leave it above the `(errors.length === 0)` check. My thinking is that if the style has errors does mapbox-gl-js always fail to render, or does it sometimes do a best effort to render the style?

Thoughts @justenPalmer?

Reply:
I thought about that, but then the app state would not get updated if the glyphs property was absent on the new style. The app would still think the glyphs property from the previous style was the active one. I think it's better just to update the state with the undefined property if the glyphs is not there, but I'm not sure the impact an undefined value could have on the interface (when testing, I didn't notice any issues).

As for moving the update glyphs and update sprite above the style validation, it seems like it would be strange to update glyphs and then have the new style not pass validation. Wouldn't that also create a state where the glyphs and sprite properties could be out of sync with the active style?

Reply:
That all makes sense, and seems well thought through. Thanks for the explanation.